PGM/src/main/java/tc/oc/pgm/picker/PickerMatchModule.java

```diff
@@ -288,29 +295,43 @@ public void closeMonitoredInventory(final InventoryCloseEvent event) {
     }
 
     @EventHandler
-    public void rightClickIcon(final ObserverInteractEvent event) {
-        if(event.getClickType() != ClickType.RIGHT) return;
-
+    public void handleClickIcon(final ObserverInteractEvent event) {
         MatchPlayer player = event.getPlayer();
-        if(!canUse(player)) return;
 
         ItemStack hand = event.getClickedItem();
         if(ItemUtils.isNothing(hand)) return;
 
         String displayName = hand.getItemMeta().getDisplayName();
         if(displayName == null) return;
 
-        if(hand.getType() == Button.JOIN.material) {
-            event.setCancelled(true);
-            if(canOpenWindow(player)) {
-                showWindow(player);
-            } else {
-                // If there is nothing to pick, just join immediately
-                jmm.requestJoin(player, JoinRequest.user());
-            }
-        } else if(hand.getType() == Button.LEAVE.material) {
-            event.setCancelled(true);
-            jmm.requestObserve(player);
+        if(hand.getType() != Button.JOIN.material) return;
+
+        switch(event.getClickType()) {
+            // Autojoin feature - Player left clicks the hat(cancels Autojoin)
+            case LEFT:
+                if(autoJoinMatchModule.shouldAutoJoin(player)) {
+                    autoJoinMatchModule.cancelAutojoin(player);
+                }
+                player.sendHotbarMessage(new Component(ChatColor.DARK_PURPLE).translate("autojoin.cancelled").bold(true));
+                break;
+            case RIGHT:
+                if(!canUse(player)) return;
+
+                if(hand.getType() == Button.JOIN.material) {
+                    event.setCancelled(true);
+                    if(canOpenWindow(player)) {
+                    showWindow(player);
+                    } else {
+                        // If there is nothing to pick, just join immediately
+                        jmm.requestJoin(player, JoinRequest.user());
+                    }
+
+                //} else if(hand.getType() == Button.LEAVE.material) {
```

Comment:
?

Reply:
Are you going to leave this here?

Reply:
pretty sure that the button doesn't do anything, so i'll leave this here at the moment until we find some use for it.

Reply:
Yes, but add a comment explaining that.

Reply:
✔️